src/content/subgames.cpp

```diff
@@ -210,40 +222,46 @@ SubgameSpec findWorldSubgame(const std::string &world_path)
 	return findSubgame(world_gameid);
 }
 
-std::set<std::string> getAvailableGameIds()
+GamePathMap getAvailableGamePaths()
 {
-	std::set<std::string> gameids;
-	std::set<std::string> gamespaths;
-	gamespaths.insert(porting::path_share + DIR_DELIM + "games");
-	gamespaths.insert(porting::path_user + DIR_DELIM + "games");
+	GamePathMap gamepaths;
```

Comment:
Perhaps rename this to `game_path_map`, `gamepaths_by_id` etc. or rename the following variable to something else like `game_search_paths`. `gamepaths` and `gamespaths` are hard to distinguish.